manager.cpp

```diff
@@ -272,8 +277,104 @@ void Manager::updateObjects(
+{
+    string systemPath =
+        "/var/lib/phosphor-inventory-manager/xyz/openbmc_project/inventory";
+
+    for (const auto& objPath : paths)
+    {
+        string absObjPath(systemPath);
+        absObjPath.append(objPath);
+
+        for (const auto& file : directory_iterator(absObjPath))
+        {
+            if (remove(file.path()))
+                cout << "File[" << file.path() << "] successfully deleted\n";
+            else
+                cout << "Error deleting this file[" << file.path() << "]\n";
+        }
+
+        // All files are deleted.
+        // Now delete directory if nothing exists in this path.
+        if (remove(absObjPath.c_str()))
+            cout << "Error deleting this object- " << absObjPath.c_str()
+                 << "\n";
+        else
+            cout << "Object successfully deleted from persistency path- "
+                 << absObjPath.c_str() << "\n";
+    }
+}
+
+void Manager::deleteObjects(map<sdbusplus::message::object_path, Object>& objs)
+{
+    auto objItr = objs.begin();
+    auto refItr = _refs.begin();
+    string absPath;
+    vector<string> paths;
+
+    auto lastKey = objs.rbegin()->first;
+    auto thisIsLastObj = false;
+
+    while (objItr != objs.end())
+    {
+        // Find the insertion point or the object to DELETE.
+        refItr = lower_bound(refItr, _refs.end(), objItr->first,
+                             compareFirst(RelPathCompare(_root)));
+
+        absPath.assign(_root);
+        absPath.append(objItr->first);
+
+        if (refItr == _refs.end() || refItr->first != absPath)
+        {
+            // requested object to delete not found/already deleted.
+            ++objItr;
+            continue;
+        }
+        else
+        {
+            // check if it has interfaces, if no then collect the object to
+            // delete and also remove it from object map
+            if (objItr->second.empty())
+            {
+                if (objItr->first == lastKey)
+                    thisIsLastObj = true;
+
+                paths.push_back(objItr->first);
+                objItr = objs.erase(objItr);
+
+                // In case last object from map is erased, then we can't access
+                // that map's iterator anymore. So break the loop.
+                if (thisIsLastObj)
+                {
+                    break;
+                }
+            }
+            else
+            {
+                ++objItr;
+                continue;
+            }
+        }
+    }
+
+    // If got something to delete.
+    if (paths.size())
+    {
+        // convert it to char* to call destroyObjects
+        vector<const char*> pathVec;
+        for (auto& p : paths)
+            pathVec.push_back(p.c_str());
+        destroyObjects(pathVec);
+
+        // Remove it from persistency also.
+        clearPersistency(paths);
+    }
+}
+
+void Manager::notify(map<sdbusplus::message::object_path, Object> objs)
 {
+    deleteObjects(objs);
```

Comment:
I think I'm missing how one can specify which objects to delete in a notify call vs which ones to update?

Reply:
we receive here a map of objects and interfaces.
The objects need to be deleted will not have any interface attached to it.
Inside deleteObjects we check for empty interface and if found we send this object to delete

Reply:
Can you add the description to the function prototype?